### DISCUSSION TITLE
Feature Request: Support for AWS credential profiles for the Bedrock integration

### DISCUSSION THREAD

**Discussion by rpg2014** [cline/cline#497] [Feature Requests]
Could you add support for AWS credential profiles? I work with several different AWS accounts, and have separate profiles configured for each one in my ~/.aws/credentials file. I'd like to specify the profile to use in the Cline settings menu.

**Comment by sarangrao** [comment]
Any possibility that this will be included anytime soon? We have multiple accounts and we use aws config profiles which have a custom credential process to manage access. Would like to use these profiles with Cline

**Comment by rpg2014** [comment]
Looks like the library being used ([@anthropic-ai/bedrock-sdk](https://www.npmjs.com/package/@anthropic-ai/bedrock-sdk)) doesn't support profiles other than the default one. I can look into migrating [this class](https://github.com/cline/cline/blob/main/src/api/providers/bedrock.ts) to the AWS SDK.  It would be a hefty dependency, but it should be ok.

**Reply by sarangrao** [reply]
That will work if you are able to refer to the ~/.aws/config file and have the ability to add the aws profile as a parameter.

**Comment by Sonic-Ninja** [comment]
There may be an easy work around for this. If you look at how the provider in the "@anthropic-ai/bedrock-sdk looks https://github.com/anthropics/anthropic-sdk-typescript/blob/main/packages/bedrock-sdk/src/auth.ts you may be able to let us set the AWS_PROFILE in the env and then don't provide any credentials except region when initiaizing the client. Courtesy Cline :)

The auth.ts file works in the following way:

The getAuthHeaders function accepts:

A request object (RequestInit)
AuthProps containing url, regionName, and optional AWS credentials
Key Implementation Details:

It uses fromNodeProviderChain() which checks credentials in this order:
Environment variables
**SSO credentials from ~/.aws/credentials**
Web Identity credentials
Shared INI file (~/.aws/credentials)
ECS credentials
EC2 Instance Metadata credentials

The interesting part is in the withTempEnv function:

It temporarily sets AWS credential environment variables
Then restores the previous environment after credentials are loaded
This means the credential chain will first check these environment variables
To use AWS SSO without code changes, you can:

```
const client = new BedrockClient({
  regionName: 'us-west-2'
  // Don't provide awsAccessKey/awsSecretKey
});
```
This works because:

When AWS_PROFILE is set, the Node provider chain prioritizes the profile credentials
The temporary environment variable logic in auth.ts won't override anything since no explicit credentials are provided
The SSO credentials from your profile will be used automatically
This leverages the existing credential chain without requiring code changes to the SDK.

**Comment by JTaylor-myenergi** [comment]
If the whole config could be set to just come from the AWS CLI profiles that would be useful. It's almost essential if you are using Bedrock and using SSO.

**Comment by bebbo-zz** [comment]
I see similar comments made from other users and multiple requests raised. I think for now there would be just a checkbox required, that you can proceed without entering any credentials and the code would not set any credentials for the AWS commands, but use the default (which is then filled from the current session).
